joomla/joomla-cms
# [RFC] Polyfill js for legacy browsers (adds events and classlist polyfills)

### Summary of Changes

As discussed in https://github.com/joomla/joomla-cms/pull/11585 this adds the ability to polyfill js events in older browsers (ex: IE8).
### Testing Instructions
- Apply https://github.com/joomla/joomla-cms/pull/11289 and them this patch
- Add this vanilla js events code in isis index.php

``` php
JHtml::_('behavior.polyfill', 'event', 'lt IE 9');
JHtml::_('behavior.polyfill', 'classlist', 'lt IE 11');

$this->addScriptDeclaration("
document.addEventListener('DOMContentLoaded', function() {

    this.getElementById('status').addEventListener('click', function() {
        alert('clicked the status bar');
    });

    this.querySelector('.container-logo').addEventListener('click', function() {
        alert('clicked the logo');
    });

    alert(this.body.classList);
});
");
```
- Refresh an admin page you will get an alert message with the body element classes
- Also, as the code says click the Joomla login in the admin and the admin status bar and check if you have a js alert.
- Test in all Joomla 3.x supported browsers. https://docs.joomla.org/Joomla_Browser_Support
- Code review
### Documentation Changes Required

Don't know. Usage examples:

``` php
JHtml::_('behavior.polyfill', 'event');
JHtml::_('behavior.polyfill', 'event', 'lt IE 9');
JHtml::_('behavior.polyfill', 'classlist', 'lt IE 11');
JHtml::_('behavior.polyfill', array('classlist'), 'lt IE 11');
JHtml::_('behavior.polyfill', array('event', 'classlist'));
JHtml::_('behavior.polyfill', array('event', 'classlist'), 'lte IE 11');
```
### Notes

Mantainers, this PR depends on https://github.com/joomla/joomla-cms/pull/11585

@Fedik @dgt41 is this it?
